app/models/piece.rb

```diff
@@ -100,4 +100,22 @@ def is_obstructed_diagonally(piece_destination)
       end 
     end 
   end 
+
+  def move_to!(x_new, y_new)
+    x_current = piece.position_x
+    y_current = piece.position_y
+    x_destination = position_x
+    y_destination = position_y
```

Comment:
@drewsef-916 still not sure what `piece` is referring to... do you mean `self`? Like accessing its own attributes?

Reply:
yes this is indeed `self` -- good to note and refactor :)